Add Faction and Ship Types to the data trees with FFG ID's

thelogicdump opened the pull request:
This patch adds both faction information and ship types to the data tree with the real name, xws name and FFG id.

They have also been added to the automatic FFG ID -> xws mapper file.

thelogicdump commented:
Also apologies if this pull request was done incorrectly. I haven't done many updates to other GitHub pull requests before and wasn't sure if there was a better way. I will delete if there is a better way.

guidokessels commented on data/ships/ships.json:
```diff
@@ -0,0 +1,227 @@
+  {
+    "name": "Modified YT-1300 Light Freighter",
+    "xws": "modifiedyt1300lightfreighter",
+    "ffg": "1"
```
I'm not sure this file is really needed. I think we should add the `xws` and `ffg` fields to json files in `data/pilots/<faction>/<ship>.json`.

thelogicdump replied:
The main reason I added this file is because we are starting to see more cross over ships such as the Tie/ln and Y-Wing. This way we can consolidate all of the ship specific features such as the dials, attack values etc removing the need to store them in the pilots "section" which could open to duplication of data and other potential issues.

guidokessels replied:
I'd prefer this info in the other ship json files for now. That way if someone wants to load all ship data they don't have to load this file and the others.

thelogicdump replied:
Cool :) No issues with this at all and it sounds reasonable.

guidokessels commented on data/stats/stats.json:
```diff
@@ -0,0 +1,66 @@
+[
```
I'm not sure what this file could be used for?

thelogicdump replied:
This file (like most of the others) are mainly about formalising specific definitions with in each ship. It also provides a link between the FFG definition and the XWS definition of a ships statistics in this particular case.

EG When you look at the Tie/ln fighter the XWS contains the following statistics about a ship. The "type" is just a string that must be common across all ships and if it needs to change for some reasons the definition (eg and if we wish to add further definition we would to find and adjust every instance. By centralising them around an xws "tag" we can then store things locally. It also allows us to validate our XWS against the FFG source for each ship easily.

EG XWS ---
"stats": [
    { "arc": "Front Arc", "type": "attack", "value": 2 },
    { "type": "agility", "value": 3 },
    { "type": "hull", "value": 3 }
],

EG FFG ---
"statistics": [
                {
                    "id": 7000,
                    "statistic_id": 10,
                    "value": "2",
                    "recurring": false
                },
                {
                    "id": 7001,
                    "statistic_id": 1,
                    "value": "3",
                    "recurring": false
                },
                {
                    "id": 7002,
                    "statistic_id": 2,
                    "value": "3",
                    "recurring": false
                }
            ],

guidokessels replied:
That makes sense.

guidokessels replied:
The only comment I have then is about the naming of the arc. They're a bit different in xwing-data

guidokessels commented on data/ffg-xws.json:
```diff
@@ -398,5 +398,82 @@
+    "3": "shields",
+    "4": "force",
+    "7": "charge",
+    "8": "dualturretarc",
```
We use `Double Turret Arc` in xwing-data

guidokessels commented on data/ffg-xws.json:
```diff
@@ -398,5 +398,82 @@
+    "9": "fullfrontarc",
+    "10": "frontarc",
+    "11": "frontbullseyearc",
+    "12": "mobilearc",
```
Interesting that FFG still calls it `mobile arc`. We call it `Single Turret Arc` in xwing-data, which I believe is what they call it in the rules reference.

thelogicdump replied:
Yep, I just checked the rules reference and will update these arcs names

thelogicdump replied:
And it appears that I had changed the source after running the script initially. But that has been updated now.

guidokessels commented on data/ffg-xws.json:
```diff
@@ -398,5 +398,82 @@
+    "8": "dualturretarc",
+    "9": "fullfrontarc",
+    "10": "frontarc",
+    "11": "frontbullseyearc",
```
This is just `Bullseye Arc` in xwing-data (no `Front`)

thelogicdump commented:
I have updated the 3 incorrect arc names (Now Bullseye Arc, Double Turret Arc and Single Turret Arc). When I get the chance I will continue porting the other linkage entries.

guidokessels commented:
Great stuff!

guidokessels commented:
I've merged #61 so you can just point this PR at `master`.

thelogicdump commented:
Done & Ready to merge.

guidokessels commented:
@thelogicdump The `data/ships/ships.json` file is still there. I thought we agreed to merge that data into the `data/pilots/...` files?

guidokessels commented:
Thanks, I see your point. But I think I'd rather drop the file for now until we see a need for it. It's easier to add it later than to remove it, as we have no idea who might be using it.

If you don't have the time I can do the change. I'd like to add slots to the mapping file as well, and was waiting for this PR to be merged to do it :)

Thanks again for the contributions!

thelogicdump commented:
I have moved the ship specific "xws" and "ffg" to the individual ship files and removed ships/ships.json

The ffg2xws file generates an identical ffg-wxs.json file both before and after the movements.

guidokessels commented:
Thanks for the changes @thelogicdump. I was travelling which is why it took a while before I could merge this.

guidokessels commented:
I'll probably update the `ffg` values to be numbers instead of strings. But will do that in a new PR.
